lib/collection/src/shards/replica_set.rs

```diff
@@ -166,7 +200,7 @@ pub struct ShardReplicaSet {
     /// List is checked on each consensus round and submitted to the consensus.
     /// If the state of the peer is changed in the consensus, it is removed from the list.
     /// Update and read operations are not performed on the peers marked as dead.
-    locally_disabled_peers: parking_lot::RwLock<HashSet<PeerId>>,
+    locally_disabled_peers: parking_lot::RwLock<HashMap<PeerId, parking_lot::Mutex<SyncState>>>,
```

Comment:
WDYT about changing the whole `HashSet` into custom structure? Otherwise looks like too much internals are exposed